src/javascripts/Services/mainTreeElementService.js

```diff
@@ -125,6 +126,11 @@ ngapp.service('mainTreeElementService', function(editModalFactory, errorService,
             errorService.try(() => scope.$emit('openModal', 'edit', modalOptions));
         };
 
+        scope.addMasters = function(node) {
+            let modalOptions = node;
```

Comment:
feels a bit weird to set the modalOptions to the node directly like this.  I think this would be cleaner if modalOptions was an object with a `node` property.  (in that case you can also omit this separate variable and just pass the object directly to the function).

Reply:
you could also, alternatively, pass only the node attributes that the modal needs to have access to.  this would make it more flexible for use in other parts of the application (I think the only thing you need to pass is the node's handle, right?)